app/scripts/lib/createDupeReqFilterStream.test.ts

```diff
@@ -0,0 +1,315 @@
+import { Transform } from 'readable-stream';
+import type { JsonRpcRequest } from '@metamask/utils';
+import createDupeReqFilterStream, {
+  THREE_MINUTES,
+} from './createDupeReqFilterStream';
+
+function createTestStream(output: JsonRpcRequest[] = []) {
+  const transformStream = createDupeReqFilterStream();
+  const testOutStream = new Transform({
+    transform: (chunk: JsonRpcRequest, _, cb) => {
+      output.push(chunk);
+      cb();
+    },
+    objectMode: true,
+  });
+
+  transformStream.pipe(testOutStream);
+
+  return transformStream;
+}
+
+function runStreamTest(
+  requests: JsonRpcRequest[] = [],
+  advanceTimersTime = 10,
+) {
+  return new Promise((resolve, reject) => {
+    const output: JsonRpcRequest[] = [];
+    const testStream = createTestStream(output);
+
+    testStream
+      .on('finish', () => resolve(output))
+      .on('error', (err) => reject(err));
+
+    requests.forEach((request) => testStream.write(request));
+    testStream.end();
+
+    jest.advanceTimersByTime(advanceTimersTime);
+  });
+}
+
+describe('createDupeReqFilterStream', () => {
```

Comment:
(could) use a similar pattern to [`streams-utils` tests](https://github.com/MetaMask/metamask-extension/pull/24533/files#diff-2a5d1a92774c0fb88530dd20b5b5140f62b0ca423d5034c680e8d538a087df9fR48), if we want to ensure this works with streams instantiated from other implementations.

Reply:
I don't _think_ we will ever need to use this with streams instantiated with other implementations, unless you, Mark or Erik can suggest a use case / need

Reply:
Do we _know_ that we don't already have versions mixed without full control of everything being single-implementation? (I'm not sure).

Also useful to ensure compatibility and prevent regressions when switching implementation (e.g. pending https://github.com/MetaMask/metamask-extension/pull/21934)

Reply:
fair enough, I can update the tests to handle those cases.

Reply:
Done in 44ac712dd2